fontforge/python.c

```diff
@@ -1783,26 +1799,28 @@ static PyFF_Point *PyFFPoint_Parse(PyObject *args, bool dup, bool always) {
     on = true;
     sel = false;
     interp = false;
-    if ( !PyArg_ParseTuple( args, "(ddii)", &x, &y, &on, &sel )) {
+    type = typemap_to_python[pt_corner];
+    if ( !PyArg_ParseTuple( args, "(ddi)|ii", &x, &y, &on, &type, &sel )) {
 	PyErr_Clear();
-	if ( !PyArg_ParseTuple( args, "(ddi)|i", &x, &y, &on, &sel )) {
+	if ( !PyArg_ParseTuple( args, "(dd)|iii", &x, &y, &on, &type, &sel )) {
 	    PyErr_Clear();
-	    if ( !PyArg_ParseTuple( args, "(dd)|ii", &x, &y, &on, &sel )) {
+	    if ( !PyArg_ParseTuple( args, "dd|iiii", &x, &y, &on, &type, &sel, &interp )) {
 		PyErr_Clear();
-		if ( !PyArg_ParseTuple( args, "dd|iii", &x, &y, &on, &sel, &interp )) {
-		    PyErr_Clear();
-		    if ( PyType_IsSubtype(&PyFF_PointType, Py_TYPE(args)) ) { 
-			p = (PyFF_Point *) args;
-		    } else if ( !always ) {
-			return( NULL );
-		    }
+		if ( PyType_IsSubtype(&PyFF_PointType, Py_TYPE(args)) ) { 
+		    p = (PyFF_Point *) args;
+		} else if ( !always ) {
+		    return( NULL );
```

Comment:
The rough equivalent of this in the last release was:
```
static PyObject *PyFFContour_InsertPoint(PyFF_Contour *self, PyObject *args) {
    double x,y;
    PyFF_Point *p=NULL;
    int i, on, pos;

    x = y = 0.0; pos = -1; on = true;
    if ( !PyArg_ParseTuple( args, "(ddi)|i", &x, &y, &on, &pos )) {
	PyErr_Clear();
	if ( !PyArg_ParseTuple( args, "(dd)|ii", &x, &y, &on, &pos )) {
	    PyErr_Clear();
	    if ( !PyArg_ParseTuple( args, "dd|ii", &x, &y, &on, &pos )) {
		PyErr_Clear();
		if ( !PyArg_ParseTuple( args, "O|i", &p, &pos ) ||
		     !PyType_IsSubtype(&PyFF_PointType, Py_TYPE(p)) )
		    return( NULL );
	    }
	}
    }
```
I explained in an earlier request why specifying `pos` this way was bad, but that that's an issue specific to `InsertPoint()` not relevant here. What is relevant is that this new code retains the subtuple (x, y, on_curve) and subtuple (x, y) options, but eliminates the rest, matching the previous code. The new `type` and `selected` properties can only be set at the "top level". `interpolated` can be set in the "flat" option to make the Pickler happy, but we won't document that.